interfaces/apparmor/backend.go

```diff
@@ -414,16 +414,36 @@ func addUpdateNSProfile(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 }
 
 func addContent(securityTag string, snapInfo *snap.Info, opts interfaces.ConfinementOptions, snippetForTag string, content map[string]*osutil.FileState) {
-	var policy string
+	// Normally we use a specific apparmor template for all snap programs.
+	policy := defaultTemplate
+	ignoreSnippets := false
+	// Classic confinement (unless overridden by JailMode) has a dedicated
+	// permissive template that applies a strict, but very open, policy.
+	if opts.Classic && !opts.JailMode {
+		policy = classicTemplate
+		ignoreSnippets = true
+	}
 	// When partial AppArmor is detected, use the classic template for now. We could
 	// use devmode, but that could generate confusing log entries for users running
 	// snaps on systems with partial AppArmor support.
-	level := release.AppArmorLevel()
-	if level == release.PartialAppArmor || (opts.Classic && !opts.JailMode) {
-		policy = classicTemplate
-	} else {
-		policy = defaultTemplate
+	if level := release.AppArmorLevel(); level == release.PartialAppArmor {
+		// As a special exception, for openSUSE Tumbleweed which ships Linux
+		// 4.16, do not downgrade confinement template.
+		//
+		// We don't want to do this in general yet because older versions of
+		// the kernel did now provide backwards compatible interpretation of a
+		// confinement so the meaning of the template would change across
+		// kernel versions and we have not validated that the current template
+		// is operational on older kernels.
+		if !release.DistroLike("opensuse-tumbleweed") {
```

Comment:
The logic is correct here but the comment only talks about Tumbleweed. I think it would be better to say:
```
		// By default, downgrade confinement to the classic template when
		// partial AppArmor support is detected. We don't want to use strict
		// in general yet because older versions of the kernel did not
		// provide backwards compatible interpretation of confinement
		// so the meaning of the template would change across kernel
		// versions and we have not validated that the current template
		// is operational on older kernels.
		//
		// As a special exception, for openSUSE Tumbleweed which ships Linux
		// 4.16, do not downgrade the confinement template.
```